qutebrowser/browser/commands.py

```diff
@@ -225,8 +225,17 @@ def tab_close(self, left=False, right=False, opposite=False, count=None):
             count: The tab index to close, or None
         """
         tab = self._cntwidget(count)
+        result = True
         if tab is None:
             return
+
+        if tab.pin is True:
+            result = message.ask("Are you sure you want to close a pinned tab?",
+                    mode=usertypes.PromptMode.yesno, default=False)
```

Comment:
This askes a blocking question, which means a local Qt eventloop will be running. While this is unavoidable sometimes, the async functions should be used whenever possible (`message.confirm_async` in this case).

This means:

- Split everything after this question in a separate private method (you can probably just call it `_tab_close`)
- If the tab is not pinned, just call `_tab_close` directly
- If it is pinned, call `message.confirm_async` with `yes_action=self._tab_close`